dataactbroker/handlers/fileHandler.py

```diff
@@ -505,20 +510,89 @@ def startGenerationJob(self, submission_id, file_type):
                 job.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
                 valJob.start_date = datetime.strptime(start_date,"%m/%d/%Y").date()
                 valJob.end_date = datetime.strptime(end_date,"%m/%d/%Y").date()
+                # Generate random uuid and store generation task
+                task_key = uuid4()
+                task = FileGenerationTask(generation_task_key = task_key, submission_id = submission_id, file_type_id = jobDb.getFileTypeId(file_type_name))
+                jobDb.session.add(task)
+
                 jobDb.session.commit()
             except ValueError as e:
                 # Date was not in expected format
                 exc = ResponseException(str(e),StatusCode.CLIENT_ERROR,ValueError)
                 return False, JsonResponse.error(exc, exc.status, url = "", start = "", end = "",  file_type = file_type)
-            get_url = CONFIG_BROKER["".join([file_type_name, "_url"])].format(cgac_code, start_date, end_date)
-            jq.generate_d_file.delay(get_url, user_id, job.job_id, InterfaceHolder, timestamped_name, self.isLocal)
+            # Create file D API URL with dates and callback URL
+            callback = "https://{}:{}/v1/complete_generation/{}/".format(CONFIG_SERVICES["broker_api_host"], CONFIG_SERVICES["broker_api_port"],task_key)
+            get_url = CONFIG_BROKER["".join([file_type_name, "_url"])].format(cgac_code, start_date, end_date, callback)
+            self.call_d_file_api(get_url)
         else:
             # TODO add generate calls for E and F
             jobDb.markJobStatus(job.job_id,"finished")
             pass
 
         return True, None
 
+    def get_xml_response_content(self, api_url):
+        """ Retrieve XML Response from the provided API url """
+        return requests.get(api_url, verify=False).text
+
+    def call_d_file_api(self, api_url):
+        job_manager = self.interfaces.jobDb
+
+        xml_response = self.get_xml_response_content(api_url)
+
+    def download_file(self, local_file_path, file_url):
+        """ Download a file locally from the specified URL """
+        with open(local_file_path, "w") as file:
+            # get request
+            response = requests.get(file_url)
+            # write to file
+            response.encoding = "utf-8"
+            file.write(response.text)
+
+    def get_lines_from_csv(self, file_path):
+        """ Retrieve all lines from specified CSV file """
+        lines = []
+        with open(file_path) as file:
+            for line in reader(file):
+                lines.append(line)
+        return lines
+
+    def load_d_file(self, url, timestamped_name, job_id, isLocal):
+        """ Pull D file from specified URL and write to S3 """
```

Comment:
Do we need to download the file before writing it to S3 (if I'm reading the code correctly, it doesn't look like we're doing anything with the local copy).